### DISCUSSION TITLE
Mods not working for me

### DISCUSSION THREAD

**Discussion by FatOrangutan** [ramensoftware/windhawk#16] [Q&A] [answered]
I'm on Windhawk v0.9.2 beta and Windows 11 Home 21H2 (22000.856).

I've tried:

1) System tray > Windhawk > Exit then running it from Program Files\Windhawk, both as Admin and without
2) Restart the WindHawk service
3) Restarting Windows Explorer
4) Restarting Windows

The app UI itself seems fine - everything loads, individual mods can be fetched, enabled and disabled. Creating UI\data and running VSCodium.exe also works. Enabling the log in Windhawk shows up this:

> INFO IExtensionHostStarter.start() took 259 ms.
log.ts:313   ERR [Extension Host] (node:17736) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `VSCodium --trace-deprecation ...` to show where the warning was created)
console.ts:137 [Extension Host] (node:17736) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.(Use `VSCodium --trace-deprecation ...` to show where the warning was created)
I @ console.ts:137
console.ts:137 [Extension Host] Git installation not found.
I @ console.ts:137
ipc.ts:409 [Violation] 'setTimeout' handler took 118ms
releaseNotesEditor.ts:223 [Violation] 'setInterval' handler took 151ms
[Violation] 'setTimeout' handler took 105ms
[Violation] 'setTimeout' handler took 98ms

<details><summary>Screenshots showing log and the 3 mods enabled</summary>
 
![image](https://user-images.githubusercontent.com/8336638/188327084-74cae133-2f2e-4dfc-a34a-ab027f4c82bc.png)

</details>

Despite the enabled mods, the taskbar is indistinguishable, both in terms of appearance and behaviour. The issue is not llimited to the taskbar mods, 'Chrome/Edge scroll tabs with mouse wheel' doesn't seem to work either.

Would appreciate any help on this.

**Comment by m417z** [comment, marked as answer]
The log in the screenshot has nothing unusual.

Let's start with the 'Chrome/Edge scroll tabs with mouse wheel' mod which is simpler. Please try the following:

* <s>Open Windhawk.
* Install the 'Chrome/Edge scroll tabs with mouse wheel' mod.
* Disable the mod.
* Fork the mod.
* Enable logging.
* Click on Show Log Output.
* Click on Compile Mod.</s>

_**Update:** If you found this comment while trying to solve a similar problem, please refer to the new way of collecting logs: https://github.com/ramensoftware/windhawk/discussions/44#discussioncomment-4951005_

You should see log output at the bottom of the window. Send me the log that you're getting. Let me know if you need help with one of the steps.

It's possible that you won't see logs at all. Let me know if that happens and I'll tell you how to enable logging for the Windhawk engine.

**Reply by FatOrangutan** [reply]
Not quite so sure I did it right, but I'm getting:

DebugViewConsole 1.8.0.102
Listening for OutputDebugString messages...
01:50:56.494 15496 msedge.exe [WH] [local@chrome-wheel-scroll-tabs-fork] [287:Wh_ModInit]: Init
01:50:56.494 1604 msedge.exe [WH] [local@chrome-wheel-scroll-tabs-fork] [287:Wh_ModInit]: Init
01:50:56.504 8436 msedge.exe [WH] [local@chrome-wheel-scroll-tabs-fork] [287:Wh_ModInit]: Init

**Reply by m417z** [reply]
Looks like you did it right. From the logs, it seems that the mod was loaded successfully, but no browser windows were found. A log message that says "Browser window created: ..." was supposed to be printed, but for some reason it wasn't. I'm not sure whether it's a problem that's specific to this mod, but let's try running it with more logs.

Please try the following code which has more logs:
https://gist.github.com/m417z/4a2f5a1c6d3e1f54e6e02f74cf93ff4f/raw/8c04241bc1b66c475d92bc1d323320e28c703ece/chrome-wheel-scroll-tabs-fork.wh.cpp
Replace the code that you see after clicking Edit with the code above, then click Compile again, and post the logs that you get. Note that the logs may contain window titles, so please remove any personal data before posting.

**Reply by FatOrangutan** [reply]
i got this:

<details>
  <summary>Expand</summary>

DebugViewConsole 1.8.0.102
Listening for OutputDebugString messages...
10:31:36.170 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [293:Wh_ModInit]: Init, cmdline: "C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe" --type=crashpad-handler "--user-data-dir=C:\Users\US\AppData\Local\Microsoft\Edge\User Data" /prefetch:7 --monitor-self-annotation=ptype=crashpad-handler "--database=C:\Users\US\AppData\Local\Microsoft\Edge\User Data\Crashpad" --annotation=IsOfficialBuild=1 --annotation=channel= --annotation=chromium-version=105.0.5195.96 "--annotation=exe=C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe" --annotation=plat=Win64 "--annotation=prod=Microsoft Edge" --annotation=ver=105.0.1343.27 --initial-client-data=0x23c,0x240,0x244,0x218,0x254,0x7ffe22e75850,0x7ffe22e75860,0x7ffe22e75870
10:31:36.170 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [298:Wh_ModInit]: Wh_SetFunctionHook: 1
10:31:36.170 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 000C0E74
10:31:36.171 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: 
10:31:36.172 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [293:Wh_ModInit]: Init, cmdline: "C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe" --no-startup-window /prefetch:5
10:31:36.172 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [298:Wh_ModInit]: Wh_SetFunctionHook: 1
10:31:36.172 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 00040DB6
10:31:36.172 948 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [293:Wh_ModInit]: Init, cmdline: "C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe" --type=utility --utility-sub-type=network.mojom.NetworkService --lang=en-US --service-sandbox-type=none --mojo-platform-channel-handle=2100 --field-trial-handle=2240,i,18360888737670971005,12675731554331367613,131072 --disable-features=OverscrollHistoryNavigation,msExperimentalScrolling /prefetch:3
10:31:36.172 948 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [298:Wh_ModInit]: Wh_SetFunctionHook: 1
10:31:36.173 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: Default IME
10:31:36.173 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [301:Wh_ModInit]: EnumWindows: 1
10:31:36.173 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [303:Wh_ModInit]: Init completed
10:31:36.175 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 00050DA6
10:31:36.175 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: 
10:31:36.175 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 00060E1C
10:31:36.176 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: 
10:31:36.176 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 00220C06
10:31:36.176 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: 
10:31:36.176 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 00180DE8
10:31:36.176 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: 
10:31:36.178 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 00110D06
10:31:36.178 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: Default IME
10:31:36.178 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [301:Wh_ModInit]: EnumWindows: 1
10:31:36.178 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [303:Wh_ModInit]: Init completed
10:31:36.179 948 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [301:Wh_ModInit]: EnumWindows: 1
10:31:36.179 948 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [303:Wh_ModInit]: Init completed

</details>

**Reply by m417z** [reply]
The mod loads and enumerates all Edge windows. Here are your logs for the main Edge process:
```
10:31:36.172 14384 msedge.exe [293:Wh_ModInit]: Init, cmdline: "C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe" --no-startup-window /prefetch:5
10:31:36.172 14384 msedge.exe [298:Wh_ModInit]: Wh_SetFunctionHook: 1
10:31:36.175 14384 msedge.exe [209:InitialEnumBrowserWindowsFunc]: Window found: 00050DA6
10:31:36.175 14384 msedge.exe [213:InitialEnumBrowserWindowsFunc]: Title:
10:31:36.175 14384 msedge.exe [209:InitialEnumBrowserWindowsFunc]: Window found: 00060E1C
10:31:36.176 14384 msedge.exe [213:InitialEnumBrowserWindowsFunc]: Title:
10:31:36.176 14384 msedge.exe [209:InitialEnumBrowserWindowsFunc]: Window found: 00220C06
10:31:36.176 14384 msedge.exe [213:InitialEnumBrowserWindowsFunc]: Title:
10:31:36.176 14384 msedge.exe [209:InitialEnumBrowserWindowsFunc]: Window found: 00180DE8
10:31:36.176 14384 msedge.exe [213:InitialEnumBrowserWindowsFunc]: Title:
10:31:36.178 14384 msedge.exe [209:InitialEnumBrowserWindowsFunc]: Window found: 00110D06
10:31:36.178 14384 msedge.exe [213:InitialEnumBrowserWindowsFunc]: Title: Default IME
10:31:36.178 14384 msedge.exe [301:Wh_ModInit]: EnumWindows: 1
10:31:36.178 14384 msedge.exe [303:Wh_ModInit]: Init completed
```

Here are my logs for the main Edge process:
```
09:38:15.027 10408 msedge.exe [293:Wh_ModInit]: Init, cmdline: "C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe" --no-startup-window /prefetch:5
09:38:15.027 10408 msedge.exe [298:Wh_ModInit]: Wh_SetFunctionHook: 1
09:38:15.027 10408 msedge.exe [209:InitialEnumBrowserWindowsFunc]: Window found: 01720E3A
09:38:15.028 10408 msedge.exe [213:InitialEnumBrowserWindowsFunc]: Title: 
09:38:15.028 10408 msedge.exe [209:InitialEnumBrowserWindowsFunc]: Window found: 00AA1A50
09:38:15.028 10408 msedge.exe [213:InitialEnumBrowserWindowsFunc]: Title: New tab - Profile 1 - Microsoft? Edge
09:38:15.028 10408 msedge.exe [216:InitialEnumBrowserWindowsFunc]: Browser window found: 00AA1A50
09:38:15.028 10408 msedge.exe [209:InitialEnumBrowserWindowsFunc]: Window found: 01C61892
09:38:15.028 10408 msedge.exe [213:InitialEnumBrowserWindowsFunc]: Title: 
09:38:15.028 10408 msedge.exe [209:InitialEnumBrowserWindowsFunc]: Window found: 00590FC0
09:38:15.028 10408 msedge.exe [213:InitialEnumBrowserWindowsFunc]: Title: 
09:38:15.028 10408 msedge.exe [209:InitialEnumBrowserWindowsFunc]: Window found: 0032156A
09:38:15.029 10408 msedge.exe [213:InitialEnumBrowserWindowsFunc]: Title: 
09:38:15.029 10408 msedge.exe [209:InitialEnumBrowserWindowsFunc]: Window found: 00C61648
09:38:15.029 10408 msedge.exe [213:InitialEnumBrowserWindowsFunc]: Title: 
09:38:15.030 10408 msedge.exe [209:InitialEnumBrowserWindowsFunc]: Window found: 00CE0DB6
09:38:15.030 10408 msedge.exe [213:InitialEnumBrowserWindowsFunc]: Title: MSCTFIME UI
09:38:15.030 10408 msedge.exe [209:InitialEnumBrowserWindowsFunc]: Window found: 00C01B54
09:38:15.030 10408 msedge.exe [213:InitialEnumBrowserWindowsFunc]: Title: Default IME
09:38:15.030 10408 msedge.exe [301:Wh_ModInit]: EnumWindows: 1
09:38:15.030 10408 msedge.exe [303:Wh_ModInit]: Init completed
```

On your system, for some reason there are no windows with a title except for the "Default IME" system window. I'd assume that you had at least one browser window open when you enabled the mod, right? If yes, then I really don't know why it's missing in the enumeration.

**Reply by FatOrangutan** [reply]
>I'd assume that you had at least one browser window open when you enabled the mod, right?

Yes but it was Firefox. Here's the log with Edge open:

```
DebugViewConsole 1.8.0.102
Listening for OutputDebugString messages...
12:31:39.339 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [293:Wh_ModInit]: Init, cmdline: "C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe" --no-startup-window /prefetch:5
12:31:39.339 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [298:Wh_ModInit]: Wh_SetFunctionHook: 1
12:31:39.339 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [293:Wh_ModInit]: Init, cmdline: "C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe" --type=crashpad-handler "--user-data-dir=C:\Users\US\AppData\Local\Microsoft\Edge\User Data" /prefetch:7 --monitor-self-annotation=ptype=crashpad-handler "--database=C:\Users\US\AppData\Local\Microsoft\Edge\User Data\Crashpad" --annotation=IsOfficialBuild=1 --annotation=channel= --annotation=chromium-version=105.0.5195.96 "--annotation=exe=C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe" --annotation=plat=Win64 "--annotation=prod=Microsoft Edge" --annotation=ver=105.0.1343.27 --initial-client-data=0x23c,0x240,0x244,0x218,0x254,0x7ffe22e75850,0x7ffe22e75860,0x7ffe22e75870
12:31:39.339 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [298:Wh_ModInit]: Wh_SetFunctionHook: 1
12:31:39.339 948 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [293:Wh_ModInit]: Init, cmdline: "C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe" --type=utility --utility-sub-type=network.mojom.NetworkService --lang=en-US --service-sandbox-type=none --mojo-platform-channel-handle=2100 --field-trial-handle=2240,i,18360888737670971005,12675731554331367613,131072 --disable-features=OverscrollHistoryNavigation,msExperimentalScrolling /prefetch:3
12:31:39.339 948 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [298:Wh_ModInit]: Wh_SetFunctionHook: 1
12:31:39.341 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 000F0EC4
12:31:39.341 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: 
12:31:39.341 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 00050DA6
12:31:39.342 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: 
12:31:39.342 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 00060E1C
12:31:39.342 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: 
12:31:39.342 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 00220C06
12:31:39.342 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 000C0E74
12:31:39.342 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: 
12:31:39.342 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: 
12:31:39.342 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 00180DE8
12:31:39.342 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: 
12:31:39.344 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 00170E68
12:31:39.345 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: windhawk - Google Search and 1 more page - Profile 1 - Microsoft? Edge
12:31:39.345 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [216:InitialEnumBrowserWindowsFunc]: Browser window found: 00170E68
12:31:39.345 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 00040DB6
12:31:39.345 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: Default IME
12:31:39.345 948 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [301:Wh_ModInit]: EnumWindows: 1
12:31:39.345 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [301:Wh_ModInit]: EnumWindows: 1
12:31:39.345 948 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [303:Wh_ModInit]: Init completed
12:31:39.345 3788 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [303:Wh_ModInit]: Init completed
12:31:39.345 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 000A0EA6
12:31:39.346 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: MSCTFIME UI
12:31:39.346 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [209:InitialEnumBrowserWindowsFunc]: Window found: 00110D06
12:31:39.346 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [213:InitialEnumBrowserWindowsFunc]: Title: Default IME
12:31:39.346 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [301:Wh_ModInit]: EnumWindows: 1
12:31:39.346 14384 msedge.exe  [WH] [local@chrome-wheel-scroll-tabs-fork] [303:Wh_ModInit]: Init completed
```

Looks like it did log the Edge window and its tabs.

**Reply by m417z** [reply]
That seems to be the correct log. And now, if you open several tabs and scroll over the tabs with the mouse wheel, does it work? See the mod gif for an illustration: https://windhawk.net/mods/chrome-wheel-scroll-tabs

**Reply by FatOrangutan** [reply]
It.. does, actually. I have no clue why it didn't when I tried it multiple times earlier.

Going by the scrolling-theme, I tried out the 'Taskbar Volume Control' mod, but couldn't get it to work. The log was:

```
DebugViewConsole 1.8.0.102
Listening for OutputDebugString messages...
14:44:25.457 9200 explorer.exe  [WH] [local@taskbar-volume-control-fork] [1787:Wh_ModInit]: Init
14:44:25.458 14460 explorer.exe  [WH] [local@taskbar-volume-control-fork] [1787:Wh_ModInit]: Init
```

I tried out 'Taskbar Labels' as well and it says:
```
DebugViewConsole 1.8.0.102
Listening for OutputDebugString messages...
14:59:50.056 14460 explorer.exe  [WH] [local@taskbar-labels-fork] [1713:Wh_ModInit]: >
14:59:50.056 9200 explorer.exe  [WH] [local@taskbar-labels-fork] [1713:Wh_ModInit]: >
```

**Reply by m417z** [reply]
Let's try Taskbar Volume Control with extra logs:
https://gist.githubusercontent.com/m417z/c27789b476328048ad3d1dc5c443a1cc/raw/762e992b3a27f1e73a660dd31a6d60602a27451a/taskbar-volume-control-fork.wh.cpp

Please compile this version, then minimize Windhawk (required because it's running as admin), then scroll your mouse over the tray area, then restore Windhawk and post the log that you got.

**Reply by FatOrangutan** [reply]
I got this:

```
DebugViewConsole 1.8.0.102
Listening for OutputDebugString messages...
00:41:16.334 9200 explorer.exe  [WH] [local@taskbar-volume-control-fork] [1792:Wh_ModInit]: Init
00:41:16.334 9200 explorer.exe  [WH] [local@taskbar-volume-control-fork] [178:WindowsVersionInit]: nMajor=10, nMinor=0, nBuild=22000, nQFE=832
00:41:16.334 9200 explorer.exe  [WH] [local@taskbar-volume-control-fork] [1804:Wh_ModInit]: Wh_SetFunctionHook(): 1
00:41:16.363 14460 explorer.exe  [WH] [local@taskbar-volume-control-fork] [1792:Wh_ModInit]: Init
00:41:16.363 14460 explorer.exe  [WH] [local@taskbar-volume-control-fork] [178:WindowsVersionInit]: nMajor=10, nMinor=0, nBuild=22000, nQFE=832
00:41:16.364 14460 explorer.exe  [WH] [local@taskbar-volume-control-fork] [1804:Wh_ModInit]: Wh_SetFunctionHook(): 1
00:41:16.364 14460 explorer.exe  [WH] [local@taskbar-volume-control-fork] [1808:Wh_ModInit]: GetClassInfo() Shell_TrayWnd
00:41:16.367 14460 explorer.exe  [WH] [local@taskbar-volume-control-fork] [1811:Wh_ModInit]: hWnd=5027a
```

BTW, something that I should have stated earlier is that I'm not using a mouse, but a precision touchpad.

And lastly, I really appreciate your help and prompt replies; I'm sorry it usually takes a while for me to get back to you.

**Reply by m417z** [reply]
The log looks fine. The touchpad is probably the reason. See the discussion here:
https://tweaker.userecho.com/topics/826-scroll-on-trackpadtouchpad-doesnt-trigger-mouse-wheel-options

You can try the "pinch to zoom" trick, or you can try attaching a mouse just to make sure it's not a problem with the mod.

If that's indeed it, we can move on to the Taskbar Labels mod. Let me know :)

**Reply by FatOrangutan** [reply]
I tried 'Pinch to zoom' but it didn't work and I don't have access to a physical mouse. It seems to be a known issue with touchpads, apparently there isn't much that can be done about it and I didn't really intend to use this feature, so I'm not particularly concerned.

>If that's indeed it, we can move on to the Taskbar Labels mod. Let me know :)

Yes please!

**Reply by m417z** [reply]
You know what to do :)
https://gist.github.com/m417z/947314e1b29276a357ba5703b8c36dff/raw/6d4da98ed389df7684f941c9bd1668a69f91ddd1/taskbar-labels-fork.wh.cpp

**Reply by FatOrangutan** [reply]
It logged:

```
DebugViewConsole 1.8.0.102
Listening for OutputDebugString messages...
02:08:47.186 14460 explorer.exe  [WH] [local@taskbar-labels-fork] [1720:Wh_ModInit]: >
02:08:47.186 14460 explorer.exe  [WH] [local@taskbar-labels-fork] [1550:HookTaskbarDllSymbols]: >
02:08:47.187 14460 explorer.exe  [WH] [local@taskbar-labels-fork] [1508:HookSymbols]: >
02:08:47.187 9200 explorer.exe  [WH] [local@taskbar-labels-fork] [1720:Wh_ModInit]: >
02:08:47.188 9200 explorer.exe  [WH] [local@taskbar-labels-fork] [1550:HookTaskbarDllSymbols]: >
02:08:47.188 9200 explorer.exe  [WH] [local@taskbar-labels-fork] [1508:HookSymbols]: >
02:08:47.345 14460 explorer.exe  [WH] [local@taskbar-labels-fork] [1513:HookSymbols]: Wh_FindFirstSymbol() failed
02:08:47.347 14460 explorer.exe  [WH] [local@taskbar-labels-fork] [1725:Wh_ModInit]: HookTaskbarDllSymbols() failed
02:08:47.349 9200 explorer.exe  [WH] [local@taskbar-labels-fork] [1513:HookSymbols]: Wh_FindFirstSymbol() failed
02:08:47.349 9200 explorer.exe  [WH] [local@taskbar-labels-fork] [1725:Wh_ModInit]: HookTaskbarDllSymbols() failed
```

And again something odd happened: with the fork enabled, the taskbar labels weren't showing up, but then I disabled the fork, re-enabled the original and it just worked! I have no clue how though: before I created this issue, I must have tried enabling-disabling the mod a whole bunch of times. Is the log indicative of symbols download failing? I had granted both ui/vscodium.exe and windhawk.exe complete access to the web (in my firewall, [SimpleWall](https://github.com/henrypp/simplewall)). Is there some other executable that perhaps needs it too?

**Reply by m417z** [reply]
explorer.exe is the process that is doing the network request in this case. And in general, the Windhawk engine is doing the network requests to download the symbol from the target process. The symbols are downloaded to `C:\ProgramData\Windhawk\Engine\Symbols`. You can disable the mod, remove the symbols, and experiment with it and with your firewall.

**Reply by FatOrangutan** [reply]
Will try these things out. Thanks a lot for your time and patience!
